frontend/dockerfile/dockerfile2llb/convert.go

```diff
@@ -1616,6 +1616,62 @@ func prefixCommand(ds *dispatchState, str string, prefixPlatform bool, platform
 	return out + str
 }
 
+// formatTargetPlatform formats a secondary platform string for cross compilation cases
+func formatTargetPlatform(base ocispecs.Platform, target *ocispecs.Platform) string {
+	if target == nil {
+		return ""
+	}
+	if target.OS == "" {
+		target.OS = base.OS
+	}
+	if target.Architecture == "" {
+		target.Architecture = base.Architecture
+	}
+	p := platforms.Normalize(*target)
+
+	if p.OS == base.OS && p.Architecture != base.Architecture {
+		archVariant := p.Architecture
+		if p.Variant != "" {
+			archVariant += "/" + p.Variant
+		}
+		return "->" + archVariant
+	}
+	if p.OS != base.OS {
+		return "->" + platforms.Format(p)
+	}
+	return ""
+}
+
+// platformFromEnv returns defined platforms based on TARGET* environment variables
+func platformFromEnv(env []string) *ocispecs.Platform {
+	var p ocispecs.Platform
+	var set bool
+	for _, v := range env {
+		parts := strings.SplitN(v, "=", 2)
+		switch parts[0] {
+		case "TARGETPLATFORM":
+			p, err := platforms.Parse(parts[1])
+			if err != nil {
+				continue
```

Comment:
Why ignore this err?

Reply:
It's a convention. A Dockerfile might be just using `ARG TARGETPLATFORM` with a completely custom value definition. We shouldn't fail the build in this case if we don't understand this value.

I can add a comment for that if you like.